# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ShareableCanvasRenderer.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f gfx/layers/ShareableCanvasRenderer.h
--- a/gfx/layers/ShareableCanvasRenderer.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/gfx/layers/ShareableCanvasRenderer.h	Wed Jun 10 22:21:02 2020 +0000
@@ -8,7 +8,7 @@
 #define GFX_SHAREABLECANVASRENDERER_H
 
 #include "CompositorTypes.h"
-#include "CopyableCanvasRenderer.h"
+#include "CanvasRenderer.h"
 #include "mozilla/layers/CanvasClient.h"
 
 namespace mozilla {
@@ -22,42 +22,38 @@
 
 namespace layers {
 
-class ShareableCanvasRenderer : public CopyableCanvasRenderer {
-  typedef CanvasClient::CanvasClientType CanvasClientType;
+class ShareableCanvasRenderer : public CanvasRenderer {
+  friend class CanvasClient2D;
+  friend class CanvasClientSharedSurface;
+
+ protected:
+  RefPtr<CanvasClient> mCanvasClient;
+
+ private:
+  layers::SurfaceDescriptor mFrontBufferDesc;
+  RefPtr<TextureClient> mFrontBufferFromDesc;
 
  public:
   ShareableCanvasRenderer();
   virtual ~ShareableCanvasRenderer();
 
  public:
-  void Initialize(const CanvasInitializeData& aData) override;
+  void Initialize(const CanvasRendererData&) override;
 
   virtual CompositableForwarder* GetForwarder() = 0;
 
   virtual bool CreateCompositable() = 0;
 
   void ClearCachedResources() override;
-  void Destroy() override;
+  void DisconnectClient() override;
 
   void UpdateCompositableClient();
 
-  const TextureFlags& Flags() const { return mFlags; }
-
   CanvasClient* GetCanvasClient() { return mCanvasClient; }
 
- protected:
-  bool UpdateTarget(gfx::DrawTarget* aDestTarget);
-
-  CanvasClientType GetCanvasClientType();
-
-  RefPtr<CanvasClient> mCanvasClient;
-
-  UniquePtr<gl::SurfaceFactory> mFactory;
-
-  TextureFlags mFlags;
-
-  friend class CanvasClient2D;
-  friend class CanvasClientSharedSurface;
+ private:
+  RefPtr<TextureClient> GetFrontBufferFromDesc(const layers::SurfaceDescriptor&,
+                                               TextureFlags);
 };
 
 }  // namespace layers